File: doc/src/scripts/SubOrbitalFlight.cpp
#include <iostream>
#include <chrono>
#include <thread>
#include <krpc.hpp>
#include <krpc/services/space_center.hpp>

int main() {
  krpc::Client conn = krpc::connect("Sub-orbital flight");
  krpc::services::SpaceCenter space_center(&conn);

  auto vessel = space_center.active_vessel();

  vessel.auto_pilot().target_pitch_and_heading(90, 90);
  vessel.auto_pilot().engage();
  vessel.control().set_throttle(1);
  std::this_thread::sleep_for(std::chrono::seconds(1));

  std::cout << "Launch!" << std::endl;
  vessel.control().activate_next_stage();

  while (vessel.resources().amount("SolidFuel") > 0.1)
    std::this_thread::sleep_for(std::chrono::seconds(1));
  std::cout << "Booster separation" << std::endl;
  vessel.control().activate_next_stage();

  while (vessel.flight().mean_altitude() < 10000)
    std::this_thread::sleep_for(std::chrono::seconds(1));

  std::cout << "Gravity turn" << std::endl;
  vessel.auto_pilot().target_pitch_and_heading(60, 90);

  while (vessel.orbit().apoapsis_altitude() < 100000)
    std::this_thread::sleep_for(std::chrono::seconds(1));
  std::cout << "Launch stage separation" << std::endl;
  vessel.control().set_throttle(0);
  std::this_thread::sleep_for(std::chrono::seconds(1));
  vessel.control().activate_next_stage();
  vessel.auto_pilot().disengage();

  while (vessel.flight().surface_altitude() > 1000)
    std::this_thread::sleep_for(std::chrono::seconds(1));
  vessel.control().activate_next_stage();

  while (vessel.flight(vessel.orbit().body().reference_frame()).vertical_speed() < -0.1) {
    std::cout << "Altitude = " << vessel.flight().surface_altitude() << " meters" << std::endl;
    std::this_thread::sleep_for(std::chrono::seconds(1));
  }
  std::cout << "Landed!" << std::endl;
}

File: doc/src/scripts/SubOrbitalFlight.cs
using System;
using KRPC.Client;
using KRPC.Client.Services.SpaceCenter;

class SubOrbitalFlight
{
    public static void Main ()
    {
        var conn = new Connection ("Sub-orbital flight");

        var vessel = conn.SpaceCenter ().ActiveVessel;

        vessel.AutoPilot.TargetPitchAndHeading (90, 90);
        vessel.AutoPilot.Engage ();
        vessel.Control.Throttle = 1;
        System.Threading.Thread.Sleep (1000);

        Console.WriteLine ("Launch!");
        vessel.Control.ActivateNextStage ();

        while (vessel.Resources.Amount("SolidFuel") > 0.1)
            System.Threading.Thread.Sleep (1000);
        Console.WriteLine ("Booster separation");
        vessel.Control.ActivateNextStage ();

        while (vessel.Flight ().MeanAltitude < 10000)
            System.Threading.Thread.Sleep (1000);

        Console.WriteLine ("Gravity turn");
        vessel.AutoPilot.TargetPitchAndHeading (60, 90);

        while (vessel.Orbit.ApoapsisAltitude < 100000)
            System.Threading.Thread.Sleep (1000);
        Console.WriteLine ("Launch stage separation");
        vessel.Control.Throttle = 0;
        System.Threading.Thread.Sleep (1000);
        vessel.Control.ActivateNextStage ();
        vessel.AutoPilot.Disengage ();

        while (vessel.Flight ().SurfaceAltitude > 1000)
            System.Threading.Thread.Sleep (1000);
        vessel.Control.ActivateNextStage ();

        while (vessel.Flight (vessel.Orbit.Body.ReferenceFrame).VerticalSpeed < -0.1) {
            Console.WriteLine ("Altitude = {0:F1} meters", vessel.Flight ().SurfaceAltitude);
            System.Threading.Thread.Sleep (1000);
        }
        Console.WriteLine ("Landed!");
        conn.Dispose();
    }
}
